docs/examples/tests/test_mlflow_utilities.py

```diff
@@ -0,0 +1,53 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+
+"""
+Snippets and tests used in the "MLFlow Utilities" tutorial.
+"""
+
+import pickle
+from pathlib import Path
+from unittest.mock import Mock
+
+from pytest import MonkeyPatch
+
+from orquestra import sdk
+
+mlflow = Mock()
+my_model = Mock()
+my_model.state_dict.return_value = {"state": "awesome"}
+
+
+class Snippets:
+    @staticmethod
+    def artifacts_dir_snippet():
+        artifacts_dir: Path = sdk.mlflow.get_temp_artifacts_dir()
+
+        artifact_path = artifacts_dir / "final_state_dict.pickle"
+        with artifact_path.open("wb"):
+            pickle.dumps(my_model.state_dict())
+        mlflow.log_artifact(artifact_path)
+
+        # </snippet>
+        return artifacts_dir
+
+
+class TestSnippets:
+    @staticmethod
+    def test_artifact_dir_local(monkeypatch, tmp_path):
+        monkeypatch.setattr("os.environ", {})
+        monkeypatch.setattr(
+            sdk.mlflow._connection_utils, "DEFAULT_TEMP_ARTIFACTS_DIR", tmp_path
+        )
+
+        assert Snippets.artifacts_dir_snippet() == tmp_path
+
+    @staticmethod
+    def test_artifact_dir_env_var_override(monkeypatch: MonkeyPatch, tmp_path):
+        monkeypatch.setattr("os.environ", {"ORQ_MLFLOW_ARTIFACTS_DIR": str(tmp_path)})
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion (non-blocking):** ‏You can use [`monkeypatch.setenv(...)`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.setenv)